### PR TITLE
feat: add hate speech binary metric

### PR DESCRIPTION
**Pull request by AzizCode92**
Following the ongoing work in this PR https://github.com/TonicAI/tonic_validate/pull/69 and the suggestion to add more metrics here https://github.com/TonicAI/tonic_validate/issues/73 I added a binary metric for hate speech.
Let me know if this would make sense to you. thanks!

### CODE REVIEW

**Comment by akamor** [comment]
Thanks @AzizCode92.  We'll take a look at this shortly. 

In the meantime, how did you find our project?  Would love to learn where folks are finding out about us.

**Comment by AzizCode92** [comment]
Thanks, @akamor.
I found out about you guys when reading HackerNews, where someone mentioned tonic_validate. I was looking for methods to evaluate a rag system.

**Comment by akamor** [comment]
> Thanks, @akamor. I found out about you guys when reading HackerNews, where someone mentioned tonic_validate. I was looking for methods to evaluate a rag system.

Thats great. Glad you found us.

File: tonic_validate/utils/llm_calls.py
Comment (ethan-tonic) [inline tonic_validate/utils/llm_calls.py:10]: Minor nit, the spacing here doesn't match the rest of the file. It also doesn't pass the formatter we use (ruff). Can you undo the spacing changes here?

Comment (AzizCode92) [reply tonic_validate/utils/llm_calls.py]: Done.

**Comment by ethan-tonic** [comment]
Also, I'll merge this once this pr https://github.com/TonicAI/tonic_validate/pull/69 goes in. Mainly because it seems some of the code in this PR is based off the PR I linked. I had to fix some bugs in that other PR and there could be more. So once that's merged I assume update this branch and then merge it if everything is good

**Comment by AzizCode92** [comment]
@ethan-tonic Hi, my PR is ready for review.

**Review by ethan-tonic — APPROVED** [review]

**Pull request merged**
